src/universal/modules/welcome/containers/Welcome/Welcome.js

```diff
@@ -62,46 +90,55 @@ export default class WelcomeContainer extends Component {
     const teamId = shortid.generate();
     const teamMemberId = shortid.generate();
     const {user} = getAuth();
-    dispatch(setWelcomeTeam({teamName, teamId, teamMemberId}));
+    dispatch(setWelcomeTeam({teamId, teamMemberId}));
     const createTeamOptions = {
       variables: {
         newTeam: {
           id: teamId,
           name: teamName,
-          leader: {
+          leader: [{
             id: teamMemberId,
             teamId,
             cachedUserId: user.id,
             isActive: true,
             isLead: true,
             isFacilitator: true
-          }
+          }]
         }
       }
     };
     cashay.mutate('createTeam', createTeamOptions);
+    this.nextPage();
   };
 
   onInviteTeamSubmit = data => {
     const {dispatch, welcome: {teamId}} = this.props;
-    const {invitees} = data;
+    const invitees = data.invitees.map(i =>
+      // Remove label field:
+      pick(i, 'email', 'fullName', 'task')
+    );
     const options = {
       variables: {
         teamId,
         invitees
       }
     };
-    cashay.mutate('inviteTeam', options)
+
+    cashay.mutate('inviteTeamMembers', options)
       .then(res => {
         // TODO make sure this resolves after the route changes
         console.log('inviteTeamRes', res);
         if (res.error) {
-          const {failedEmails} = JSON.parse(res.error);
```

Comment:
Ha! Abuse!

I have to admit, I like the `then()` pattern. It allows you to conveniently dispatch your own actions rather than mapping Cashay state to a prop and then possibly doing the same thing if you wanted to distribute the error to other components.

All the same, no objection to you changing how it works here. Just wanted to let you know I didn't think you committed any crimes 😉